publish.sh

```diff
@@ -2,73 +2,117 @@
 
 set -x
 
-if [[ "${CIRCLE_TAG}" == *dev ]]; then
+artifact_dir=~/artifacts/linux
+for arch in x86_64 arm64; do
+    if [ ! -f "${artifact_dir}/extension-${arch}.zip" ]; then
+        echo "${arch} extension does not exist, cannot publish."
+        exit 1;
+    fi
+done
+
+VERSION="${CIRCLE_TAG:-$(make version)}"
+# lambda layer names must match a regex: [a-zA-Z0-9-_]+)|[a-zA-Z0-9-_]+
+# turn periods into dashes and cry into your coffee
+VERSION=$(echo ${VERSION} | tr '.' '-')
+
+if [[ "${VERSION}" == *dev ]]; then
     EXTENSION_NAME="honeycomb-lambda-extension-dev"
 else
     EXTENSION_NAME="honeycomb-lambda-extension"
 fi
 
-REGIONS_NO_ARCH=(eu-north-1 us-west-1 eu-west-3 ap-northeast-2 sa-east-1 ca-central-1 af-south-1
-                ap-east-1 eu-south-1 me-south-1 ap-southeast-3 ap-northeast-3)
-REGIONS_WITH_ARCH=(ap-south-1 eu-west-2 us-east-1 eu-west-1 ap-northeast-1 ap-southeast-1
-                   ap-southeast-2 eu-central-1 us-east-2 us-west-2)
+# Region list update from AWS Lambda pricing page as of 2022/10/12
+#
+# regions with x86_64 only support
+# REGIONS_NO_ARCH=(me-central-1) # listed on pricing page, but we need grant our publishing credentials permission to publish there first
+REGIONS_NO_ARCH=()
+# Regions with x86_64 & arm64
+REGIONS_WITH_ARCH=(
```

Comment:
Showing my work on determining this new list.

* Copied the `<ul>` lists of x86_64 and arm64 supporting regions from the [AWS Lambda pricing page](https://aws.amazon.com/lambda/pricing/). (ಠ_ಠ)
* Trimmed the content down from the HTML to only the region identifiers in those lists.
* Diff'd the lists.

```shell
» diff --ignore-matching-lines --side-by-side regions-arm64.txt regions-x86_64.txt
16a17
> me-central-1
20a22,23
> us-gov-east-1
> us-gov-west-1
```

<details><summary>(fuller diff)</summary>

```shell
 » diff --side-by-side regions-arm64.txt regions-x86_64.txt
af-south-1							af-south-1
ap-east-1							ap-east-1
ap-northeast-1							ap-northeast-1
ap-northeast-2							ap-northeast-2
ap-northeast-3							ap-northeast-3
ap-south-1							ap-south-1
ap-southeast-1							ap-southeast-1
ap-southeast-2							ap-southeast-2
ap-southeast-3							ap-southeast-3
ca-central-1							ca-central-1
eu-central-1							eu-central-1
eu-north-1							eu-north-1
eu-south-1							eu-south-1
eu-west-1							eu-west-1
eu-west-2							eu-west-2
eu-west-3							eu-west-3
							      >	me-central-1
me-south-1							me-south-1
sa-east-1							sa-east-1
us-east-1							us-east-1
us-east-2							us-east-2
							      >	us-gov-east-1
							      >	us-gov-west-1
us-west-1							us-west-1
us-west-2							us-west-2
```

</details>

OK. Only Middle East (UAE) and GovCloud in the x86_64-only camp.

How does the new list of regions with arm64 support compare with our previous lists? They match.

```ruby
irb> NEW_REGIONS_WITH_ARM64 == (REGIONS_NO_ARCH + REGIONS_WITH_ARCH)
=> true
```

<details><summary>Details: Set comparisons with Ruby</summary>

```ruby
# Because Ruby is a tool near to my hand …
» irb
irb(main):001:1] REGIONS_NO_ARCH=Set.new(%w[eu-north-1 us-west-1 eu-west-3 ap-northeast-2 sa-east-1 ca-central-1 af-south-1
irb(main):002:1]                 ap-east-1 eu-south-1 me-south-1 ap-southeast-3 ap-northeast-3])
=>
#<Set:
...
irb(main):003:1] REGIONS_WITH_ARCH=Set.new(%w[ap-south-1 eu-west-2 us-east-1 eu-west-1 ap-northeast-1 ap-southeast-1
irb(main):004:1]                    ap-southeast-2 eu-central-1 us-east-2 us-west-2])
=> #<Set: {"ap-south-1", "eu-west-2", "us-east-1", "eu-west-1", "ap-northeast-1", "ap-southeast-1", "ap-southeast-2", "eu-central-1", "us-east-2", "us-west-2"}>
irb(main):005:1] NEW_REGIONS_WITH_ARM64=Set.new(%w[
irb(main):006:1]   af-south-1
irb(main):007:1]   ap-east-1
irb(main):008:1]   ap-northeast-1
irb(main):009:1]   ap-northeast-2
irb(main):010:1]   ap-northeast-3
irb(main):011:1]   ap-south-1
irb(main):012:1]   ap-southeast-1
irb(main):013:1]   ap-southeast-2
irb(main):014:1]   ap-southeast-3
irb(main):015:1]   ca-central-1
irb(main):016:1]   eu-central-1
irb(main):017:1]   eu-north-1
irb(main):018:1]   eu-south-1
irb(main):019:1]   eu-west-1
irb(main):020:1]   eu-west-2
irb(main):021:1]   eu-west-3
irb(main):022:1]   me-south-1
irb(main):023:1]   sa-east-1
irb(main):024:1]   us-east-1
irb(main):025:1]   us-east-2
irb(main):026:1]   us-west-1
irb(main):027:1]   us-west-2])
=>
#<Set:
...
irb(main):028:0> NEW_REGIONS_WITH_ARM64 == (REGIONS_NO_ARCH + REGIONS_WITH_ARCH)
=> true
```

</details>